Algorithmia/datafile.py

```diff
@@ -186,23 +186,35 @@ def putJson(self, data):
         # Post to data api
         jsonElement = json.dumps(data)
         result = localPutHelper(self.path, jsonElement)
-        if 'error' in result: raise DataApiError(result['error']['message'])
-        else: return self
+        if 'error' in result:
+            raiseApiError(result)
+        else:
+            return self
 
     def putFile(self, path):
         result = localPutHelper(path, self.path)
-        if 'error' in result: raise DataApiError(result['error']['message'])
-        else: return self
+        if 'error' in result:
+            raiseApiError(result)
+        else:
+            return self
 
     def delete(self):
         try:
             os.remove(self.path)
             return True
-        except: raise DataApiError('Failed to delete local file ' + self.path)
+        except:
+            raise DataApiError('Failed to delete local file ' + self.path)
 
 def localPutHelper(path, contents):
     try:
         with open(path, 'wb') as f:
             f.write(contents)
             return dict(status='success')
     except Exception as e: return dict(error=str(e))
+
+def raiseApiError(result):
+    if 'error' in result:
```

Comment:
Since `if 'error' in result:` is already checked before calling `raiseApiError` function, maybe we can have that conditional removed either from within this function or from the code that's calling it? Otherwise, we're checking this key twice.

Reply:
Yea, I was on the fence about that. The problem is that I didn't want to pass `result['error']` directly because we may want to use information that is outside the error property. So I wanted to send the whole result.

Then, once I'm in the raiseApiError function, I have no guarantee that `'error' in result` except for my own carefulness when coding elsewhere. Maybe I've been writing too much typescript but I felt like it was just better to be safe there. Especially since this PR is in response to overly "trusting" error code.

Reply:
Yeah I totally get what you mean, we won't have any guarantee of the caller having made this check before calling this function so the function itself cannot rely on that. 
K then, I think it's acceptable in our case. The rest looks good to me!